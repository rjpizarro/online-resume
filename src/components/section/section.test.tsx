import React from 'react'
import renderer from 'react-test-renderer'
import { render } from "@testing-library/react"

import Section from './index'
import { Divider } from "@material-ui/core"

describe.only("Section", () => {
  it("render normally with required props", () => {
    const tree = renderer
      .create(
        <Section>
          <div>
            <span>section</span>
          </div>
        </Section>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("render normally with title and divider props", () => {
    const tree = renderer
      .create(
        <Section divider title="Section Title">
          <div>
            <span>section</span>
          </div>
        </Section>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("add a Divider component when receive a divider prop", () => {
    const { getByTestId } = render(
      <Section divider>
        <div>
          <span>section</span>
        </div>
      </Section>
    )

    expect(getByTestId("divider")).toBeInTheDocument()
    expect(getByTestId("divider")).toMatchSnapshot()
  })

  it("add a title when receive a title prop", () => {
    const { getByTestId, debug, getByText } = render(
      <Section title="Section Title">
        <div>
          <span>section</span>
        </div>
      </Section>
    )

    expect(getByTestId("title")).toBeInTheDocument()
    expect(getByTestId("title")).toHaveTextContent("Section Title")
  })
})