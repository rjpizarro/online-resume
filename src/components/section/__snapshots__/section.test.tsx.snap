// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section add a Divider component when receive a divider prop 1`] = `
<hr
  class="MuiDivider-root"
  data-testid="divider"
/>
`;

exports[`Section render normally with required props 1`] = `
<div
  className="makeStyles-container-1"
>
  <section
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div>
      <span>
        section
      </span>
    </div>
  </section>
</div>
`;

exports[`Section render normally with title and divider props 1`] = `
<div
  className="makeStyles-container-1"
>
  <h2
    className="MuiTypography-root makeStyles-sectionTitle-2 MuiTypography-h2"
    data-testid="title"
  >
    Section Title
  </h2>
  <hr
    className="MuiDivider-root"
    data-testid="divider"
  />
  <section
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div>
      <span>
        section
      </span>
    </div>
  </section>
</div>
`;
